beacon-chain/rpc/core/validator.go

```diff
@@ -401,7 +401,19 @@ func (s *Service) GetAttestationData(
 	if err != nil {
 		return nil, &RpcError{Reason: Internal, Err: errors.Wrap(err, "could not get target root")}
 	}
-	justifiedCheckpoint := s.FinalizedFetcher.CurrentJustifiedCheckpt()
+
+	headState, err := s.HeadFetcher.HeadState(ctx)
```

Comment:
you should fetch the read only version of the head state otherwise this will do a copy and blow up the memory of nodes running with many keys.

Reply:
It's just one copy per slot because of the cache. But I'll change it to read-only because that's always better